internal/controllers/reserve-controller/reserve.go

```diff
@@ -0,0 +1,169 @@
+package reserve_controller
+
+import (
+	"context"
+	"encoding/json"
+	user_reserve "example/restaurant-reserved/internal/infrastructure/repositories/user-reserve"
+	models "example/restaurant-reserved/internal/share-domain/reserve/reserve-model"
+	"example/restaurant-reserved/tools/response"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+func GetReserves() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		var reserves []models.ReserveModel
+		defer cancel()
```

Comment:
swip create var and defer it will be better

ctx, cancel := context.something()
defer cancel()